### PR TITLE
update library target with includes

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,11 @@ install(FILES "inc/azure_macro_utils/macro_utils.h" DESTINATION "${CMAKE_INSTALL
 
 if("${CMAKE_VERSION}" VERSION_GREATER 3.0.0)
     add_library(azure_macro_utils_c INTERFACE)
+    target_include_directories(azure_macro_utils_c
+        INTERFACE
+            $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/inc>
+            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/azure_macro_utils>
+    )
     install(TARGETS azure_macro_utils_c EXPORT azure_macro_utils_cTargets
         INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```
Comment (dcristoloveanu) [inline CMakeLists.txt]: >${CMAKE_INSTALL_INCLUDEDIR} [](start = 29, length = 27)

Is this supposed to be ${CMAKE_INSTALL_INCLUDEDIR}/azure_macro_utils

File: CMakeLists.txt [hunk 2]
```diff
@@ -24,6 +24,11 @@ install(FILES "inc/azure_macro_utils/macro_utils.h" DESTINATION "${CMAKE_INSTALL
 
 if("${CMAKE_VERSION}" VERSION_GREATER 3.0.0)
     add_library(azure_macro_utils_c INTERFACE)
+    target_include_directories(azure_macro_utils_c
+        INTERFACE
+            $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/inc>
+            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/azure_macro_utils>
+    )
     install(TARGETS azure_macro_utils_c EXPORT azure_macro_utils_cTargets
```
Comment (dcristoloveanu) [inline CMakeLists.txt]: Can we drop the install thing for the love of everything? It was only needed for Edge v1. Who else needs it?
And Vcpkg should have it's own CMakeLists (think CMakeLists.vcpkg, which can include CMakeLists.Txt if it wants but let's please separate this out as it's becoming a nightmare.

File: CMakeLists.txt (hunk 1 above)
Comment (danewalton) [reply CMakeLists.txt]: ah i think you're right. line 23 threw me for a loop saying it is installing at `/azure_macro_utils`

File: CMakeLists.txt (hunk 2 above)
Comment (danewalton) [reply CMakeLists.txt]: that would have to be a different PR. not having this generator expression breaks the cmake generation/build. but i mostly agree.

File: CMakeLists.txt (hunk 1 above)
Comment (dcristoloveanu) [reply CMakeLists.txt]: This is a classic example of hairs :-).

ln. 23 has some install FILES (with the header). I missed that one completely while reviewing btw. I was looking around asking "where is that install(FILES specifying that the header is installed".

Then we have some generator expressions in 27. Then targets in 32.

Then some more export and files.

Bleh.

Now let's imagine how nice would this look if these install statements would not exist here. At all.
Cause if I am a dev it's highly unlikely for me to care about this user scenario of installing it.

And for that user scenario we should have this separate file where the install statements are nicer (possibly even done with a common function that you can shove into the  C Build tools repo. [https://github.com/Azure/azure-c-build-tools](https://github.com/Azure/azure-c-build-tools).

I know. I am beating a dead horse.

---
In reply to: [320403840](https://github.com/Azure/azure-macro-utils-c/pull/21#discussion_r320403840) [](ancestors = 320403840)

Comment (dcristoloveanu) [reply CMakeLists.txt]: Actually I am trying to extort from you the promise of changing this to separate the install part :-).

---
In reply to: [320410477](https://github.com/Azure/azure-macro-utils-c/pull/21#discussion_r320410477) [](ancestors = 320410477,320403840)

Comment (dcristoloveanu) [reply CMakeLists.txt]: Even if it's a separate PR

---
In reply to: [320410748](https://github.com/Azure/azure-macro-utils-c/pull/21#discussion_r320410748) [](ancestors = 320410748,320410477,320403840)

Comment (danewalton) [reply CMakeLists.txt]: yea i think things could be improved for sure. I'm curious if you could have a generic install function which can just use the properties of a given target to install it. I'd like the imagine that would be possible.

---
In reply to: [320410850](https://github.com/Azure/azure-macro-utils-c/pull/21#discussion_r320410850) [](ancestors = 320410850,320410748,320410477,320403840)

**Pull request merged**